terraform/tls-web-application/README.md

```diff
@@ -0,0 +1,59 @@
+# DNSimple TLS & Terraform
+
+An example library using Terraform to configure a DNSimple registered domain with TLS using Let's Encrypt.
+
+The demo web application is built using vite + react. The web application build is copied into the docker image defined in `terraform/main.tf` along with the resulting SSL/TLS certificate data and required nginx configurations.
+
+## Requirements
+
+- Docker
+- NodeJS
+- Terraform
+
+## DevContainer
+
+The provided `.devcontainer` configures Debian Bullseye with:
+
+- common-utils
+- docker-outside-of-docker
+- node
+- terraform
+
+## Setup
+
+### Build Web Application
+
+Requires NodeJS + preferred package manager (examples using yarn)
+
+1. `cd react-app`
+2. `yarn && yarn build`
+
+### Terraform Variables
+
+Create `terraform/terraform.tfvars` and set the following variables:
+
+```
+dnsimple_account = "{ACCOUNT_ID}"
+dnsimple_token = "{API_TOKEN}"
+dnsimple_domain = "{DNSIMPLE_REGISTERED_DOMAIN.TLD}"
+```
+
+See `terraform.tfvars.example` for examples.
+
+## Run Terraform
+
+1. `cd terraform`
+2. `terraform init`
+3. `terraform import dnsimple_registered_domain.domain {DNSIMPLE_REGISTERED_DOMAIN.TLD}` (Optional)
+4. `terraform plan`
+5. `terraform apply`
+
```

Comment:
I think perhaps a step is missing to add an entry in the hosts file for the domain on the machine.